youtube_dl/extractor/polskieradio.py

```diff
@@ -15,6 +16,67 @@
 )
 
 
+class PolskieRadioProgrammeIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?polskieradio\.pl/(?P<id>\d+/\d+)'
+    _TESTS = [{
+        'url': 'http://www.polskieradio.pl/7/5102,HISTORIA-ZYWA',
+        'info_dict': {
+            'id': '7/5102',
+            'title': 'HISTORIA ŻYWA',
+        },
+        'playlist_mincount': 34,
+    }, {
+        'url': 'http://www.polskieradio.pl/7/4807',
+        'info_dict': {
+            'id': '7/4807',
+            'title': 'Vademecum 1050. rocznicy Chrztu Polski'
+        },
+        'playlist_mincount': 5
+    }, {
+        'url': 'http://www.polskieradio.pl/7/129,Sygnaly-dnia?ref=source',
+        'only_matching': True
+    }]
+
+    def _get_entries_from_page_content(self, base_url, content):
+        entries = []
+
+        articles = re.findall(
+            r'<article class="ID-(\d+) article">\s+<a href="([^"]+)" class="[^"]*" title="([^"]+)">',
+            content)
+        for article in articles:
+            article_id, article_url, article_title = article
+            resolved_article_url = compat_urlparse.urljoin(base_url, article_url)
+            entries.append(self.url_result(
+                resolved_article_url,
+                ie='PolskieRadio',
+                video_id=article_id,
+                video_title=article_title))
+
+        return entries
+
+    @classmethod
+    def suitable(cls, url):
+        return False if PolskieRadioIE.suitable(url) else super(PolskieRadioProgrammeIE, cls).suitable(url)
+
+    def _real_extract(self, url):
+        programme_id = self._match_id(url)
+        webpage = self._download_webpage(url, programme_id)
+
+        title = self._html_search_regex(
+            r'<h2 id=".*_lblTitle">(.+?)</h2>',
+            webpage, 'title', fatal=False)
+        description = None
+
+        page_urls = re.findall(r'<a href="(/' + programme_id + '/Strona/\d+)"', webpage)[:-1]
```

Comment:
This approach does not work when there are too much pages to be displayed at once: http://www.polskieradio.pl/7/1551,Kultura. Correct approach is to follow next link on each page.